include/boost/unordered/detail/foa/table.hpp

```diff
@@ -128,10 +133,21 @@ class table_iterator
   template<typename> friend class table_erase_return_type;
   template<typename,typename,typename,typename> friend class table;
 
-  table_iterator(Group* pg,std::size_t n,const table_element_type* p_):
-    pc{reinterpret_cast<unsigned char*>(const_cast<group_type*>(pg))+n},
-    p{const_cast<table_element_type*>(p_)}
-    {}
+  table_iterator(group_type* pg,std::size_t n,const table_element_type* p_)
+  {
+    if(pg)
```

Comment:
There might be.

I only added the check because of a ubsan violation.

With raw pointers, our code isn't UB.

This condition is still holding for me:
```cpp
if(!pg){BOOST_ASSERT(n==0);}
```

But because of our `pointer_to` shenanigans, ubsan seems to throw an error, which is why I added the check.

Reply:
Yes, my bad assumption, this constructor can legitimately take null pointers.